src/internal/internal_rbt_generate.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2020-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "internal/internal.hh"
+#include "slate/Matrix.hh"
+
+#include <cmath>
+#include <stdlib.h>
+#include <vector>
+
+
+namespace slate {
+
+namespace internal {
+
+template<typename scalar_t>
+void rbt_fill(Matrix<scalar_t>& U, const int64_t seed)
+{
+    using real_t = blas::real_type<scalar_t>;
+
+    const int64_t d = U.n();
+    slate_assert(d == U.tileNb(0));
+    slate_assert(U.nt() == 1);
+    const real_t scale_20 = 20.0;
+
+    U.insertLocalTiles( Target::Host );
+    for (int64_t i = 0; i < U.mt(); i++) {
+        if (U.tileIsLocal(i, 0)) {
+            U.tileGetForWriting( i, 0, LayoutConvert::None );
+            Tile<scalar_t> U_i = U(i, 0);
+
+            const int64_t mb = U_i.mb();
+
+            #pragma omp task
+            {
+                int64_t iseed[4] = {(seed + i) % 4096, 578, 361, 115};
+                lapack::larnv( 2, iseed, mb*d, U_i.data() );
+
+                for (int64_t k = 0; k < d; k++) {
+                    for (int64_t jj = 0; jj < mb; jj++) {
+                        real_t U_jk = blas::real( U_i.at(jj, k) );
+                        U_i.at(jj, k) = std::exp( U_jk/scale_20 );
+                    }
+                }
+            }
+        }
+    }
+
+}
+
+template<typename scalar_t>
```

Comment:
As far as I can tell, the description of the functions here are missing, like what `d` is.